pints/_log_likelihoods.py

```diff
@@ -133,6 +133,131 @@ def __call__(self, x):
                       - np.sum(autocorr_error**2, axis=0) / (2 * sigma**2))
 
 
+class GaussianIntegratedUniformLogLikelihood(pints.ProblemLogLikelihood):
+    r"""
+    Calculates a log-likelihood assuming independent Gaussian-distributed noise
+    at each time point where :math:`\sigma\sim U(a,b)` has been integrated
+    out of the joint posterior of :math:`p(\theta,\sigma|X)`,
+    .. math::
+        \begin{align} p(\theta|X) &= \int_{0}^{\infty} p(\theta, \sigma|X)
+        \mathrm{d}\sigma\\
+        &\propto \int_{0}^{\infty} p(X|\theta, \sigma) p(\theta, \sigma)
+        \mathrm{d}\sigma,\end{align}
+    Note that this is exactly the same statistical model as
+    `pints.GaussianLogLikelihood` with a uniform prior on :math:`\sigma`.
+    A possible advantage of this log-likelihood compared with using a
+    `pints.GaussianLogLikelihood`, is that it has one fewer
+    parameter (:math:`sigma`) which may speed up convergence to the
+    posterior distribution, especially for multi-output problems which will
+    have `n_outputs` fewer parameter dimensions.
+    The log-likelihood is given in terms of the sum of squared errors:
+    .. math::
+        SSE = \sum_{i=1}^n (f_i(\\theta) - y_i)^2
+    and is given up to a normalisation constant by:
+    .. math::
+        \text{log } L = - n / 2 \text{log}(\pi) -
+            \text{log}(2 (b - a) \sqrt(2)) +
+            (1 / 2 - n / 2) \text{log}(SSE) +
+            \text{log}\\left[\Gamma((n - 1) / 2, \frac{SSE}{2 b^2}) -
+            \Gamma((n - 1) / 2, \frac{SSE}{2 a^2}) \right]
+    where :math:`\Gamma(u,v)` is the upper incomplete gamma function
+    as defined here: https://en.wikipedia.org/wiki/Incomplete_gamma_function
+    This log-likelihood is inherently a Bayesian method since it
+    assumes a uniform prior on :math:`\sigma\sim U(a,b)`. However
+    using this likelihood in optimisation routines should yield the
+    same estimates as the full `pints.GaussianLogLikelihood`.
+    Arguments:
```

Comment:
Newline before & after "Arguments:" required to get the rendered formatting right

Reply:
Done